spec/index_definition_spec.rb

```diff
@@ -24,7 +24,11 @@ class Post < ::ActiveRecord::Base ; end
   end
 
   after(:each) do
-    migration.remove_index :users, :name => 'users_login_index' if migration.index_name_exists? :users, 'users_login_index', nil
+    should_remove = Gem::Version.new(ActiveRecord::VERSION::STRING) >= Gem::Version.new('5.2') ?
```

Comment:
Would it be clearer to:
```ruby
next if Gem::Version... ? migration. ... : migration ...
```

And drop the `if should_remove` below?

Reply:
@lowjoel Or how about this?

```ruby
args = [:users, 'users_login_index']
args << nil if migration.method(:index_name_exists?).arity == 3 # or the gem version check

migrate.remove_index :users, name: 'users_login_index' if migration.index_name_exists?(*args)
```

Reply:
Also, for consistency, as you've mentioned:
```ruby
if Gem::Version.new(.....) >= ....
  after(:each) ...
else
  after(:each) ...
end
```

I think this is slightly easier to see why we're doing the fancy argument stuffing, and is easy to remove when we want to drop support for older versions. Also bonus points for being used all over the gem (if it is...)